packages/data-stores/src/reader/mutations/use-site-subscribe-mutation.ts

```diff
@@ -62,9 +64,29 @@ const useSiteSubscribeMutation = ( blog_id?: string ) => {
 
 			return response;
 		},
-		onMutate: async () => {
+		onMutate: async ( params ) => {
 			await queryClient.cancelQueries( siteSubscriptionDetailsCacheKey );
 
+			const previousSiteSubscriptions =
+				queryClient.getQueryData< SiteSubscriptionsPages >( siteSubscriptionsCacheKey );
+
+			if ( previousSiteSubscriptions ) {
+				queryClient.setQueryData( siteSubscriptionsCacheKey, {
+					...previousSiteSubscriptions,
+					pages: previousSiteSubscriptions.pages.map( ( page ) => {
+						return {
+							...page,
+							total_subscriptions: page.total_subscriptions - 1,
+							subscriptions: page.subscriptions.map( ( siteSubscription ) => ( {
+								...siteSubscription,
+								isDeleted:
+									siteSubscription.blog_ID === params.blog_id ? false : siteSubscription.isDeleted,
```

Comment:
When resubscribing, the timestamp is incorrect (due to optimistic update). Could we optimistically update the `since` timestamp to the current time when a person resubscribes around here?